src/main/scala/dependencies/model/MavenDependency.scala

```diff
@@ -1,7 +1,7 @@
 package dependencies.model
 
-case class MavenDependency(coordinates: ArtifactCoordinates,
-                           dependsOn: Set[MavenDependency])
-    extends Ordered[MavenDependency] {
-  override def compare(that: MavenDependency): Int = coordinates.compare(that.coordinates)
+case class MavenDependency[C <: ArtifactCoordinates](coordinates: C,
```

Comment:
Made this generic b/c I end up processing the union of all third-party dependencies into groupings based on details of the `coordinates` field specific to only `MavenAccountCoordinates`.